### PR TITLE
Fix globby import pruned by webpack

### LINKED ISSUE #290: vscode extension fails with error cannot find package 'globby' with latest version
### Summary

The v0.5.4 of ansible-language-server fails with error
```
Cannot find package 'globby' imported from /Users/ganeshrn/.vscode/extensions/redhat.ansible-0.8.1/out/server/src/server.js
```
when used in the vscode-ansible extension.
 
Most likely the issue is caused by https://github.com/ansible/ansible-language-server/commit/4861a9ae434c74877583056d5c8cf6e20f0995c5

### Extension version

v0.5.4

### VS Code version

1.63.2

### Ansible Version

```shell
2.12
```

### OS / Environment

mac os

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by tomaciazek**
This implements a workaround for globby import being pruned by webpack and subsequently failing client-side.
Fixes #290

### CODE REVIEW

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
Thanks @tomaciazek for fixing the issue.

After the PR is merged will rerun https://github.com/ansible/ansible-language-server/pull/289 to test vscode-ansible test-e2e testcases for unreleased (main branch) of ansible-language-server

**Review by ganeshrn — CHANGES_REQUESTED** [review]
I am still seeing the issue after following the steps here https://github.com/ansible/ansible-language-server/blob/c9915bf46369b4182e0212599fa7b65c3d2a71eb/.github/workflows/vscode.yml#L36-L48

Also the issue can be  reproduced by following below steps:
1) Running `npm pack` within `ansible-langauge-server`
2)  Installing the generated tar of ansible-language-server in the vscode-ansible repo with
 command `npm install ../ansible-language-server/ansible-ansible-language-server-0.5.4.tgz `
3) Run `npm run webpack`
4) Run `npm run vsce` 
5) Install generated vsix using `code --install-extension ansible-0.8.1.vsix --force`

**Comment by tomaciazek** [comment]
@ganeshrn Yeah, I totally forgot there is yet another issue with globby: it is calling internal node libraries using modern notation `node:*`, which seems to break with webpack too. There is a workaround for that too, which requires something like this in `vscode-ansible`'s `webpack.config.js`:
```
  externals: [
    function ({ request }, callback) {
      const nodeMatch = request.match(/^node:(.*)$/);
      if (nodeMatch) {
        // Remove the 'node:' prefix
        return callback(null, nodeMatch[1]);
      }
      // Continue without changes
      callback();
    },
    {
      vscode: "commonjs vscode", // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed
    },
  ]
```

I tested it following your procedure, and this time it would work...

But I'm starting to agree with you both - maybe this is not the right way. Even though I still think we might hit this kind of issue with some other library (should the maintainers choose to go full ESM), maybe it's enough that we have this backup solution here as a _closed_ (unmerged) PR, and just replace globby with some other tool 😉

**Comment by priyamsahoo** [comment]
The problem is with the globby compartibility. Right now, to import globby, the code needs to be compiled first (since we have a getGlobby function for it). That's the reason why we don't see any errors while running the extension in dev host mode after we do `npm run compile-withserver`. The problem starts appearing when we make an npm package of the als code.

There could be 2 solutions for it:

1. Find out a way to make dynamic imports work in npm packages. (maybe, it is not supported).
2. Replace globby with some other library that does the work for us.

**Pull request closed without merging**
